GridPoint[] gridPnt
int32 mapGridWidth
int32 mapGridHeight
